src/main/java/bio/terra/cli/context/WorkspaceContext.java

```diff
@@ -189,18 +186,85 @@ public void removeCloudResource(String name) {
   }
 
   // ====================================================
-  // Directory and file names
-  //   - current working directory: .
-  //   - persisted workspace context file: ./terra-cli/workspace_context.json
-  //   - sub-directories for tools (e.g. ./nextflow) are defined in the SupportedToolHelper
-  // sub-classes
+  // Resolving file and directory paths
+
+  /**
+   * Get the workspace directory. (i.e. the parent of the .terra directory)
+   *
+   * @return the absolute path to the workspace directory
+   */
+  @SuppressFBWarnings(
+      value = "NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE",
+      justification =
+          "An NPE would only happen here if there was an error getting the workspace context file, and an exception would be thrown in that method instead.")
+  @JsonIgnore
+  public static Path getWorkspaceDir() {
+    return getWorkspaceContextFile().getParent().getParent();
+  }
+
+  /**
+   * Get a handle for the workspace context file.
```

Comment:
handles or paths? is it absolute or relative to the current directory?

Reply:
Ah I missed these references to handles, thanks for the catch. Updated comment to indicate absolute.